common/sstuff.hh

```diff
@@ -345,6 +345,16 @@ public:
     return res;
   }
 
+  //! Reads a block of data from the socket to a block of memory
+  //! Only returns when all the data requested is available
+  int readAll(char *buffer, int bytes)
+  {
+    int res=::recv(d_socket,buffer,bytes, MSG_WAITALL);
```

Comment:
i am not sure if this is a good idea. maybe better idea would be to call ::recv until enough bytes (decrement the bytes with what has been read), until all has been received or error occurs?

Reply:
I could, but how would that be different from using MSG_WAITALL, which essentially implements exactly the same algorithm.

Reply:
hmm. MSG_WAITALL can return less than bytes. It does not guarantee that res = bytes.